src/BenchmarkDotNet.Core/Jobs/JobExtensions.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using BenchmarkDotNet.Toolchains;
+using BenchmarkDotNet.Characteristics;
+using BenchmarkDotNet.Engines;
+using BenchmarkDotNet.Environments;
+using BenchmarkDotNet.Horology;
+
+namespace BenchmarkDotNet.Jobs
+{
+    public static class JobExtensions
```

Comment:
@AndreyAkinshin 
yep, but each call will incur additional allocation due to closure on WithXxx() arg.
If that's ok - will do.